components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt

```diff
@@ -312,6 +312,8 @@ class EditToolbar internal constructor(
 
     private fun onTextChanged(text: String) {
         views.clear.isVisible = text.isNotBlank()
+        views.editActionsEnd.autoHideAction(text.isNotEmpty())
```

Comment:
It's a bit weird to me that we use `isNotBlank` above since any string with whitespaces only (ex "   ") will not show the clear button, but not sure if this is the right place to deal with it 🤷‍♂️